File: Project_24/LifeTracker/frontend/lib/models/update_meal_request.dart
import 'create_meal_request.dart';
import 'meal_type.dart';

class UpdateMealRequest {
  const UpdateMealRequest({
    required this.mealType,
    required this.mealDate,
    this.notes,
    required this.items,
  });

  final MealType mealType;
  final DateTime mealDate;
  final String? notes;
  final List<MealItemRequest> items;

  Map<String, dynamic> toJson() {
    return CreateMealRequest(
      mealType: mealType,
      mealDate: mealDate,
      notes: notes,
      items: items,
    ).toJson();
  }
}

File: Project_24/LifeTracker/backend/src/main/java/com/lifetracker/modules/meals/dto/UpdateMealRequest.java
package com.lifetracker.modules.meals.dto;

import com.lifetracker.modules.meals.enums.MealType;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;
import java.util.List;

public record UpdateMealRequest(
        @NotNull MealType mealType,
        @NotNull LocalDate mealDate,
        String notes,
        @NotEmpty @Valid List<MealItemRequest> items
) {
}
